### PR TITLE
Fix flaky titanic validator test

### LINKED ISSUE #629: Titanic validator test is flaky
The `validator.spec.ts` test that validates the titanic task is flaky, it requires a minimum accuracy of 0.5 and fails around half of the time.

How to reproduce: 
1. build discojs: `cd discojs; npm run build`
2. launch a server instance `cd server; npm run dev`
3. run the titanic test: `cd discojs/discojs-core; npx mocha -g "works for titanic"`

### PR DESCRIPTION
**Pull request by JulienVig**
Fixes #629 

When uncommenting the titanic validator test (which was previously commented because the loss was NaN), I increased the accuracy minimum without noticing that the accuracy was tested at initialization, i.e., without any training.
Therefore, I lower the validator accuracy expectations (from 0.5 to 0.3) but I add assertions on minimum training and validation accuracy (both > 0.6) in the end-to-end test.

### CODE REVIEW

**Review by tharvik — APPROVED** [review]
hoo nice, you found out why, thanks!
